### PR TITLE
[Crypto] Register local components

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
Spin-off from #5429

Registers the local components:

- `crypto.localstorage`
- `crypto.jwks`
- `crypto.kubernetes.secrets`

They are all based on the same abstract "local crypto" base component

### CODE REVIEW

File: cmd/daprd/components/crypto_azure_keyvault.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/*
+Copyright 2021 The Dapr Authors
```
Comment (mukundansundar) [inline cmd/daprd/components/crypto_azure_keyvault.go:2]: nitpick 2023 in all new files.

File: cmd/daprd/components/crypto_kubernetes_secrets.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+package components
+
+import (
+	"github.com/dapr/components-contrib/crypto/kubernetes/secrets"
+	cryptoLoader "github.com/dapr/dapr/pkg/components/crypto"
+)
+
+func init() {
+	cryptoLoader.DefaultRegistry.RegisterComponent(secrets.NewKubeSecretsCrypto, "kubernetes.secrets")
```
Comment (mukundansundar) [inline cmd/daprd/components/crypto_kubernetes_secrets.go:22]: do we need `k8s.secrets`? Can it just be `kubernetes`?

**Review by mukundansundar — COMMENTED** [review]
small nitpicks ...

File: cmd/daprd/components/crypto_azure_keyvault.go (hunk 1 above)
Comment (ItalyPaleAle) [reply cmd/daprd/components/crypto_azure_keyvault.go]: Done

**Comment by ItalyPaleAle** [comment]
Proposed names:

`crypto.local.file`
`crypto.local.jwks`
`crypto.local.kubernetes.secrets`
`crypto.azure.keyvault`

**Comment by JoshVanL** [comment]
I propose `local` -> `daprManagedKey`.

**Comment by mukundansundar** [comment]
`local` -> `native`?

**Comment by ItalyPaleAle** [comment]
I changed this PR to only have the "local" components while we agree on the naming. I'll move the Azure component to a separate PR so it can be merged in the meanwhile

**Comment by mukundansundar** [comment]
For the naming part, can we change `local` to `inmemory`? Since all ops are performed in memory and keys are available in memory. 
cc @yaron2 @artursouza @msfussell @berndverst

**Comment by ItalyPaleAle** [comment]
Let's try to get a consensus on naming!

Here are some proposed names from this thread and some other additions:

- `local`
- `inmemory`
- `builtin`
- `daprManagedKey`
- `native`
- `dapr`
- `in-dapr`
- `localkey`

**Comment by yaron2** [comment]
+1 binding on dapr. My reasoning is that component names typically point to what the component is and not to where it stores/processes the data. Theoretically, an inmemory implementation can use an external component to implement the functionality.

**Comment by ItalyPaleAle** [comment]
Ok so I've updated the PR and these are the current names as implemented:

- `crypto.dapr.localstorage` (I think `localstorage` is better than `file` because it actually watches a folder, and not a file - but I can revert)
- `crypto.dapr.kubernetes.secrets` (yes, it's a mouthful but I think `kubernetes.secrets` is required in case we need to support other K8s resources in the future)
- `crypto.dapr.jwks` (this actually watches a file, unlike `localstorage` :) - or a URL or a fixed value passed in the metadata)

**Comment by mukundansundar** [comment]
> Ok so I've updated the PR and these are the current names as implemented:
> 
> * `crypto.dapr.localstorage` (I think `localstorage` is better than `file` because it actually watches a folder, and not a file - but I can revert)
> * `crypto.dapr.kubernetes.secrets` (yes, it's a mouthful but I think `kubernetes.secrets` is required in case we need to support other K8s resources in the future)
> * `crypto.dapr.jwks` (this actually watches a file, unlike `localstorage` :) - or a URL or a fixed value passed in the metadata)

These are needed for high level crypto APIs also right @ItalyPaleAle ?

**Review by mukundansundar — APPROVED** [review]
lgtm

**Comment by ItalyPaleAle** [comment]
> These are needed for high level crypto APIs also right

Yes they are used by both low-level and high level. I had included this diagram in the proposal, hoping it clarifies:

https://github.com/dapr/proposals/blob/main/resources/0002-RCBS-Crypto-building-block/data-flow.png

**Pull request merged**